subprojects/core-api/src/main/java/org/gradle/caching/local/DirectoryBuildCache.java

```diff
@@ -81,4 +72,17 @@ public void setRemoveUnusedEntriesAfterDays(int removeUnusedEntriesAfterDays) {
 
         this.removeUnusedEntriesAfterDays = removeUnusedEntriesAfterDays;
     }
+
+    static class DirectoryAdapter {
+        @BytecodeUpgrade
+        @Nullable
+        static Object getDirectory(DirectoryBuildCache buildCache) {
+            return buildCache.getDirectory().getAsFile().getOrNull();
+        }
+
+        @BytecodeUpgrade
+        static void setDirectory(DirectoryBuildCache buildCache, @Nullable Object directory) {
+            buildCache.getDirectory().set(buildCache.getDirectory().files(directory).getSingleFile());
```

Comment:
👍  Nice trick

Reply:
What does this trick do and why is it necessary? Can we please add a comment about it in the code?